ckanext/collaborators/blueprint.py

```diff
@@ -0,0 +1,135 @@
+from flask import Blueprint
+from flask.views import MethodView
+
+from ckan.common import _, g
+
+import ckan.plugins.toolkit as toolkit
+import ckan.model as model
+import ckan.lib.navl.dictization_functions as dictization_functions
+import ckan.logic as logic
+
+
+def collaborators_read(dataset_id):
+    context = {u'model': model, u'user': toolkit.c.user}
+    data_dict = {'id': dataset_id}
+      
+    # needed to ckan_extend package/edit_base.html
+    try:    
+        g.pkg_dict = toolkit.get_action('package_show')(context, data_dict)
+    except toolkit.NotAuthorized: 
+        message = 'Unauthorized to read dataset {0}'.format(dataset_id)
+        return toolkit.abort(401, toolkit._(message))
+    except toolkit.ObjectNotFound:
+        return toolkit.abort(404, toolkit._(u'Resource not found'))
+    
+    return toolkit.render('collaborator/collaborators.html')
+
+def collaborator_delete(dataset_id, user_id):
+    context = {u'model': model, u'user': toolkit.c.user}
+
+    try:
+        toolkit.get_action('dataset_collaborator_delete')(context, {
+            'id': dataset_id,
+            'user_id': user_id
+        })
+    except toolkit.NotAuthorized: 
+        message = u'Unauthorized to read dataset {0}'.format(dataset_id)
+        return toolkit.abort(401, toolkit._(message))
+    except toolkit.ObjectNotFound:
+        return toolkit.abort(404, toolkit._(u'Resource not found'))
+
+    return toolkit.redirect_to(u'collaborators.read', dataset_id=dataset_id)
+
+class CollaboratorEditView(MethodView):
+    def post(self, dataset_id):
+        context = {u'model': model, u'user': toolkit.c.user}
+
+        try:
+            form_dict = logic.clean_dict(
+                dictization_functions.unflatten(
+                    logic.tuplize_dict(
+                        logic.parse_params(toolkit.request.form))))
+            
+            user = toolkit.get_action('user_show')(context, {
+                'id':form_dict['username']
+                })
+            
+            data_dict = {
+                'id': dataset_id,
+                'user_id': user['id'],
+                'capacity': form_dict['capacity']
+            }
+            
+            toolkit.get_action('dataset_collaborator_create')(
+                context, data_dict)
+
+        except dictization_functions.DataError:
+            return toolkit.abort(400, _(u'Integrity Error'))
+        except toolkit.NotAuthorized: 
+            message = u'Unauthorized to read dataset {0}'.format(dataset_id)
+            return toolkit.abort(401, toolkit._(message))
+        except toolkit.ObjectNotFound:
+            return toolkit.abort(404, toolkit._(u'Resource not found'))
+        except toolkit.ValidationError as e:
+            toolkit.h.flash_error(e.error_summary)
+            return toolkit.redirect_to(u'collaborators.read',
```

Comment:
This line is unnecessary as the one immediately below will be executed after leaving the `except` block

Reply:
fixed in master